Cython/Compiler/Nodes.py

```diff
@@ -1582,6 +1582,9 @@ def func_attributes(attributes):
                 elif isinstance(attr, CompilerDirectivesNode):
                     for sub_attr in func_attributes(attr.body.stats):
                         yield sub_attr
+                elif isinstance(attr, CppClassNode) and attr.attributes is not None:
+                    for sub_attr in func_attributes(attr.attributes):
+                        yield sub_attr
```

Comment:
Here we are in the AST (type) analysis part.

In a cppclass, the same way we are bubbling up functions nested inside a CompilerDirectivesNode, we are bubbling up defined methods of nested cppclass.

After two years without looking at Cython code, my understanding of this part of the PR is fragile.

From what I'm able to recall ( / bet) this is for adding nested methods definitions like "ClassA.ClassB.method1()"
into the C++ code generation pipeline. I guess without that, the nested methods are just declared, and their code is not generated in the C++ code.